tests/playwright/src/model/ai-lab-page.ts

```diff
@@ -0,0 +1,130 @@
+
+    this.startRecipeButton = this.webview.getByRole('button', { name: 'Start recipe' });
+    this.confirmStartRecipeButton = this.webview.getByRole('button', { name: /Start(\s+([A-Za-z]+\s+)+)recipe/i });
+    this.recipeStatus = this.webview.getByRole('status');
+
+    this.applicationDetailsPanel = webview.getByLabel('application details panel');
+    this.openAIAppButton = this.applicationDetailsPanel.getByRole('button', { name: 'Open AI App' });
+    this.deleteAIAppButton = this.applicationDetailsPanel.getByRole('button', { name: 'Delete AI App' });
+  }
+
+  async waitForPageLoad(): Promise<void> {
+    await playExpect(this.heading).toBeVisible();
+    await playExpect(this.navigationBar).toBeVisible();
+    await playExpect(this.recipesCatalogButton).toBeVisible();
+    await playExpect(this.runningAppsButton).toBeVisible();
+    await playExpect(this.catalogButton).toBeVisible();
+    await playExpect(this.servicesButton).toBeVisible();
+    await playExpect(this.playgroundsButton).toBeVisible();
+  }
+
+  async openRecipesCatalog(): Promise<void> {
+    await playExpect(this.recipesCatalogButton).toBeVisible();
+    await this.recipesCatalogButton.click();
+    await this.page.waitForTimeout(2000);
+    await playExpect(this.recipesCatalogPage).toBeVisible();
+    await playExpect(this.recipesCatalogContent).toBeVisible();
+    await playExpect(this.recipesCatalogNaturalLanguageProcessing).toBeVisible();
+    await playExpect(this.recipesCatalogAudio).toBeVisible();
+    await playExpect(this.recipesCatalogComputerVision).toBeVisible();
+  }
+
+  async openRecipesCatalogApp(category: Locator, appName: string): Promise<void> {
+    await this.openRecipesCatalog();
+    await category.getByRole('region', { name: 'content', exact: true }).first().getByRole('region', { name: appName, exact: true }).getByRole('button', { name: 'More details' }).click({ timeout: 5_000 });
+    await playExpect(this.webview.getByRole('heading', { name: appName })).toBeVisible();
+  }
+
+  /**
+   * @throws {Error} If the app details are not visible or Delete AI App button is not visible
+   */
+  async tryDeleteOpenApp(): Promise<void> {
+    await playExpect(this.applicationDetailsPanel).toBeVisible();
+    await this.deleteAIAppButton.click({ timeout: 5_000 });
+    try {
+      await playExpect(this.page.getByRole('dialog', { name: 'Podman AI Lab' })).toBeVisible();
+      await handleConfirmationDialog(this.page, 'Confirm');
+    } catch(error) {
+      console.warn(`Warning: Delete app confirmation dialog interaction failed, possibly app is not installed.\n\t${error}`);
+    }
+    await playExpect(this.applicationDetailsPanel).toContainText('AI App Removed', { timeout: 30_000 });
+    await playExpect(this.startRecipeButton).toBeVisible();
+  }
+
+  async startOpenApp(): Promise<void> {
+    await this.startRecipeButton.click({ timeout: 5_000 });
+    await this.confirmStartRecipeButton.click({ timeout: 5_000 });
+    try {
```

Comment:
I'm not sure I understand why this try-catch block exists here?

Reply:
that dialog used to pop up when there is an image already downloaded. It will ask you to reset, continue or cancel

Reply:
fixed in new page fragment